### PR TITLE
text-pitch-alignment

### PR DESCRIPTION
**Pull request by yhahn**
Native port of https://github.com/mapbox/mapbox-gl-js/pull/2668.
native-port

### Backlog

- [x] Render tests pass
- [x] Code review
- [x] Coordinate with https://github.com/mapbox/mapbox-gl-js/pull/2668, style spec attribute

cc @lucaswoj @jfirebaugh @mollymerp @lxbarth

### CODE REVIEW

File: src/mbgl/renderer/painter_symbol.cpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ void Painter::renderSDF(SymbolBucket &bucket,
     bool alignedWithMap = rotationAlignment == RotationAlignmentType::Map;
     float gammaScale = 1.0f;
 
-    if (alignedWithMap) {
+    if (false && alignedWithMap) {
```
Comment (yhahn) [inline src/mbgl/renderer/painter_symbol.cpp]: I'll fix this, was poking but this whole codepath can be dropped.

**Comment by brunoabinader** [comment]
Good stuff @yhahn! Do you know if this affects the recently added `mbgl::ViewportMode::FlippedY`?

**Comment by yhahn** [comment]
@jfirebaugh :pray: ready for your review when you have a chance : )

File: scripts/generate-style-code.js [hunk 2]
```diff
@@ -6,12 +6,14 @@ const spec = require('mapbox-gl-style-spec').latest;
 var parseCSSColor = require('csscolorparser').parseCSSColor;
 
 global.camelize = function (str) {
+  str = str === undefined ? 'undefined' : str;
```
Comment (jfirebaugh) [inline scripts/generate-style-code.js]: What prompted this change, and the corresponding one in `camelizeWithLeadingLowercase`?

Comment (yhahn) [inline scripts/generate-style-code.js]: This was to have `Undefined` be the value for `text-pitch-alignment` when it's not defined in the style and be able to detect when inheritance should kick in (per https://github.com/mapbox/mapbox-gl-style-spec/pull/460#issuecomment-225238097).

It's definitely the main eyebrow-raiser in this PR :smile: would love to know if there's a better way to achieve this.

Comment (jfirebaugh) [inline scripts/generate-style-code.js]: Yeah, I see the `Alignment::Undefined`. Why does a JS `undefined` value pass through `camelize` though?

**Comment by jfirebaugh** [comment]
👌

**Pull request merged**